wasm/src/lib.rs

```diff
@@ -24,13 +24,14 @@ use data_model::{
 use debug::DebugExpectExt as _;
 pub use iroha_data_model as data_model;
 pub use iroha_wasm_derive::main;
+use lol_alloc::{FreeListAllocator, LockedAllocator};
 use parity_scale_codec::{DecodeAll, Encode};
 
 pub mod debug;
 pub mod log;
 
 #[global_allocator]
-static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
```

Comment:
Looking at the docs, `LockedAllocator` is only required if multi-threading is used in wasm. With [AssumeSingleThreaded](https://docs.rs/lol_alloc/0.4.0/lol_alloc/struct.AssumeSingleThreaded.html), a few bytes may be shaved off.